lib/compress/zstdmt_compress.c

```diff
@@ -1818,6 +1835,60 @@ static int ZSTDMT_tryGetInputRange(ZSTDMT_CCtx* mtctx)
     return 1;
 }
 
+/**
+ * Searches through the input for a synchronization point. If one is found, we
+ * will instruct the caller to flush, and return the number of bytes to load.
+ * Otherwise, we will load as many bytes as possible and continue as normal.
+ */
+static size_t findSynchronizationPoint(ZSTDMT_CCtx* mtctx, ZSTD_inBuffer* input, ZSTD_EndDirective* endOp) {
+    size_t toLoad = MIN(input->size - input->pos, mtctx->targetSectionSize - mtctx->inBuff.filled);
+    if (!mtctx->params.rsyncable)
+        /* Rsync is disabled. */
+        return toLoad;
+    if (mtctx->inBuff.filled + toLoad < RSYNC_LENGTH)
+      /* Not enough to compute the hash.
```

Comment:
It effectively guarantees that a job size is a minimum of `RSYNC_LENGTH` bytes,
irrespective of the fact that it may contain, or not, a synchronization point.

I guess it's fine.
An alternative could have been to load data from `roundBuff`,
since `inBuff` is merely a slice of `roundBuff` now.
But if the only perspective is to produce tiny jobs, that does not seem a great outcome.

Reply:
Yeah, I thought about it, but it added extra complexity, for virtually no benefit.

We might miss a synchronization point in this range that would sync us up, but the probability is very low (<= 32 / 2^20), and we'll just catch the next one.